### PR TITLE
nudoku: init at 1.0.0

### PR DESCRIPTION
**Pull request by reinhardt**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/games/nudoku/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  inherit (s) name version;
+  inherit buildInputs;
+  src = fetchurl {
+    inherit (s) url sha256;
+  };
+  preInstall = ''
+    mkdir -p {share/man,bin}
+  '';
+  meta = {
+    inherit (s) version;
```
Comment (etu) [inline pkgs/games/nudoku/default.nix]: We usually don't have version in meta, so not sure if it does anything at all?

File: pkgs/games/nudoku/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    description = ''A ncurses based sudoku game'';
```
Comment (etu) [inline pkgs/games/nudoku/default.nix]: `''` are generally used for multi line strings, so regular `"` can be used instead :)

File: pkgs/games/nudoku/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+stdenv.mkDerivation {
+  inherit (s) name version;
+  inherit buildInputs;
+  src = fetchurl {
+    inherit (s) url sha256;
+  };
+  preInstall = ''
+    mkdir -p {share/man,bin}
+  '';
+  meta = {
```
Comment (etu) [inline pkgs/games/nudoku/default.nix]: We usually add `meta = with stdenv.lib; {` to make lines below more concise

File: pkgs/games/nudoku/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    inherit (s) version;
+    description = ''A ncurses based sudoku game'';
+    license = stdenv.lib.licenses.gpl3;
```
Comment (etu) [inline pkgs/games/nudoku/default.nix]: can be: `license = licenses.gpl3;`

File: pkgs/games/nudoku/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    maintainers = [ "Manuel Reinhardt <manuel.reinhardt@neon-cathedral.net>" ];
```
Comment (etu) [inline pkgs/games/nudoku/default.nix]: Should refer to an entry in the maintainer list and not be a string :)

File: pkgs/games/nudoku/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    platforms = stdenv.lib.platforms.linux;
```
Comment (etu) [inline pkgs/games/nudoku/default.nix]: `platforms = platforms.linux;`

File: pkgs/games/nudoku/default.nix [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    homepage = https://github.com/jubalh/nudoku;
```
Comment (etu) [inline pkgs/games/nudoku/default.nix]: Non-quoted URLs scheme will go away, so it should be a string.

File: pkgs/games/nudoku/default.nix [hunk 8]
```diff
@@ -0,0 +1,32 @@
+{stdenv, fetchurl, ncurses}:
+let
```
Comment (etu) [inline pkgs/games/nudoku/default.nix]: We could just drop the let-block and put the values where they belong

File: pkgs/games/nudoku/default.nix [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    version = "1.0.0";
+    name = "${baseName}-${version}";
+    url = "https://github.com/jubalh/nudoku/releases/download/${version}/${baseName}-${version}.tar.xz";
+    sha256 = "0nr2j2z07nxk70s8xnmmpzccxicf7kn5mbwby2kg6aq8paarjm8k";
+  };
+  buildInputs = [
+    ncurses
+  ];
+in
+stdenv.mkDerivation {
```
Comment (etu) [inline pkgs/games/nudoku/default.nix]: Let's add rec to this one so we can use the variables in other places.

File: pkgs/games/nudoku/default.nix [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+  inherit (s) name version;
```
Comment (etu) [inline pkgs/games/nudoku/default.nix]:
```nix
pname = "nudoku";
version = "1.0.0";
```

Name will be built from pname and version :)

File: pkgs/games/nudoku/default.nix [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+  inherit buildInputs;
```
Comment (etu) [inline pkgs/games/nudoku/default.nix]: This one can just be moved from the let block? :)

File: pkgs/games/nudoku/default.nix [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  src = fetchurl {
+    inherit (s) url sha256;
```
Comment (etu) [inline pkgs/games/nudoku/default.nix]:
```
url = "https://github.com/jubalh/nudoku/releases/download/${version}/${pname}-${version}.tar.xz";
sha256 = "0nr2j2z07nxk70s8xnmmpzccxicf7kn5mbwby2kg6aq8paarjm8k";
```

That's fine, but why not `fetchFromGitHub`? Sometimes there's good reasons to not use it (like they supply something extra in the actual releases).

**Review by etu — CHANGES_REQUESTED** [review]
Well, had I bunch of thoughts... :)

**Comment by reinhardt** [comment]
Thanks for the feedback! I copied most of the stuff from some other package, but maybe I picked a bad one. Do you happen to have a good example that I could base a new package definition on?

**Comment by reinhardt** [comment]
This was an interesting exercise, but I'm not following up on it any more.

**Pull request closed without merging**